modules/storages/app/controllers/storages/admin/oauth_clients_controller.rb

```diff
@@ -55,14 +55,21 @@ def new
   # Actually create a OAuthClient object.
   # Use service pattern to create a new OAuthClient
   # Called by: Global app/config/routes.rb to serve Web page
-  def create
+  def create # rubocop:disable Metrics/AbcSize
     service_result = ::OAuthClients::CreateService.new(user: User.current)
                                                   .call(permitted_oauth_client_params.merge(integration: @storage))
     @oauth_client = service_result.result
     if service_result.success?
       flash[:notice] = I18n.t(:notice_successful_create)
-      # admin_settings_storage_path is automagically created by Ruby routes.
-      redirect_to admin_settings_storage_path(@storage)
+      if OpenProject::FeatureDecisions.automatically_managed_project_folders_active?
+        if @storage.automatic_management_unspecified?
+          redirect_to new_admin_settings_storage_automatically_managed_project_folders_path(@storage)
+        else
+          redirect_to admin_settings_storage_path(@storage)
+        end
+      else
+        redirect_to admin_settings_storage_path(@storage)
+      end
```

Comment:
2 of the branches end up in the same place. Do we need the nested `if`s for some reason or can we simply join then?
```suggestion
      if OpenProject::FeatureDecisions.automatically_managed_project_folders_active? && @storage.automatic_management_unspecified?
          redirect_to new_admin_settings_storage_automatically_managed_project_folders_path(@storage)
      else
        redirect_to admin_settings_storage_path(@storage)
      end
```

Reply:
Given one branch is wrapped in a feature flag, that is bound to be removed- I found it more readable in it's current form and simpler to understand- albeit minimal duplication.